examples/teleport.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+"""
+Qwerty demo of quantum teleportation as described in Section 1.3.7 of Nielsen
+and Chuang. Intended to be run from the command line.
+"""
+
+from argparse import ArgumentParser
+from qwerty import *
+
+@qpu
+def teleport(secret: qubit) -> qubit:
+    alice, bob = '00' or '11'
+
+    # Now imagine Alice and Bob being separated. Alice wants
+    # to transfer a qubit state (named "secret" here) to Bob.
+
+    m_pm, m_std = secret + alice | '1' & std.flip \
+                                 | (pm + std).measure
+
+    # Imagine that Alice transmits her classical bits m_pm
+    # and m_std to Bob.
+
+    # Now using the two classical bits, Bob can recover the
+    # secret state!
+    secret_teleported = \
+        bob | (pm.flip if m_std else id) \
+            | (std.flip if m_pm else id)
+
+    return secret_teleported
```

Comment:
More up-to-date version of this: https://github.com/gt-tinker/qwerty/blob/65e618895c01c8cd55f76eb821463bd1f87ef773/qwerty_pyrt/python/qwerty/tests/integ/meta_noinfer/teleport.py#L7-L19